docs/source/relayer.rst

```diff
@@ -0,0 +1,105 @@
+Cross-chain message relayer
+===========================
+
+| This component is responsible for handling the cross-chain contract state updates. 
+|
+| It is mainly used by the Agent software when resolving requests as a result of a running challenge game
+  or for sending fill invalidation messages (L1 resolutions). However, it can also be used manually by running
+  the CLI commands.
+| If you haven't read about the role of :ref:`l1-resolutions` in Beamer, make sure you get familiar 
+| with this concept first.
+|
+| Basically, every **request fill** action creates a **proof** on the request destination chain.
+  The agent utilizes these proofs in an optimistic way. If a challenge game ever occurs, or if a
+  fill was invalid, the agent has the chance to prove that a misbehavior occured and in the case of a challenge game,
+  get rewarded for it.  
```

Comment:
To someone coming from outside, it would not be quite clear what is meant by `optimistic` above. I think we should clarify that a bit more (perhaps at the risk of repeating some things from the L1 resolution part).

E.g.
```
The agent utilizes these if and when needed. In normal operation, agent's claims will not be challenged so fill proofs are not needed, therefore the agent will not invoke the relayer.

However, if a challenge game occurs, the agent has to make use of the proof to ensure it wins the game. Typically, that is done by invoking the relayer at an appropriate moment, i.e. when the proof becomes available for execution on Ethereum L1.
```